fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1512,432 +1526,477 @@ private Bundle processBundleEntries(Bundle requestBundle, Bundle responseBundle,
-                        addLocalRefMapping(localRefMap, localIdentifier, resource);
-                    }
-                }
-            }
-
-            // Next, for PUT and DELETE requests, we need to sort the indices by the request url path value.
-            if (httpMethod.equals(HTTPVerb.PUT) || httpMethod.equals(HTTPVerb.DELETE)) {
-                sortBundleRequestEntries(requestBundle, entryIndices);
-                if (log.isLoggable(Level.FINER)) {
-                    log.finer("Sorted bundle request indices to be processed: "
-                            + entryIndices.toString());
-                }
-            }
-
-            // Now visit each of the request entries using the list of indices obtained above.
-            // Use hashmap to store both the index and the according updated response bundle entry.
-            HashMap<Integer, Bundle.Entry> responseIndexAndEntries =
-                    new HashMap<Integer, Bundle.Entry>();
-            for (Integer entryIndex : entryIndices) {
+            // Visit each of the request entries, processing those with the specified request method.
+            // Use hashmap to store both the index and the accordingly updated response bundle entry.
+            HashMap<Integer, Bundle.Entry> responseIndexAndEntries = new HashMap<Integer, Bundle.Entry>();
+            for (int entryIndex=0; entryIndex<requestBundle.getEntry().size(); ++entryIndex) {
                 Bundle.Entry requestEntry = requestBundle.getEntry().get(entryIndex);
-                Bundle.Entry responseEntry = responseBundle.getEntry().get(entryIndex);
-                Bundle.Entry.Builder responseEntryBuilder = responseEntry.toBuilder();
-
                 Bundle.Entry.Request request = requestEntry.getRequest();
+                Bundle.Entry responseEntry = responseBundle.getEntry().get(entryIndex);
                 Bundle.Entry.Response response = responseEntry.getResponse();
+                if (response.getStatus().equals(string(Integer.toString(SC_OK))) && request.getMethod().equals(httpMethod)) {
+                    // Process request entry.
+                    Bundle.Entry.Builder responseEntryBuilder = responseEntry.toBuilder();
+                    StringBuffer requestDescription = new StringBuffer();
+                    long initialTime = System.currentTimeMillis();
+                    
+                    try {
+                        FHIRUrlParser requestURL = new FHIRUrlParser(request.getUrl().getValue());
+
+                        if (log.isLoggable(Level.FINER)) {
+                            log.finer("Processing bundle request entry " + entryIndex + "; method="
+                                    + request.getMethod().getValue() + ", url="
+                                    + request.getUrl().getValue());
+                            log.finer("--> path: " + requestURL.getPath());
+                            log.finer("--> query: " + requestURL.getQuery());
```

Comment:
same with this line --- 

I suggest bracketing the log with a single quote as spaces are hard to see otherwise.

Reply:
Done.